src/Modules/OrchardCore.Commerce.ContentFields/Views/AddressFieldEditor-WesternCommonNameParts.cshtml

```diff
@@ -112,7 +110,7 @@
 <script at="Foot" depends-on="@ResourceNames.CommerceRegions">
     @if (Context.Items["commerce-regions-initialized"] == null)
     {
-        <text>commerceRegionsInitialize(@Html.Raw(JsonConvert.SerializeObject(Model.Provinces)));</text>
+        <text>commerceRegionsInitialize(@Json.Serialize(Model.Provinces));</text>
```

Comment:
```suggestion
        <text>commerceRegionsInitialize(@JConvert.SerializeObject(Model.Provinces));</text>
```

Reply:
This suggestion broke the page so I had to revert it here and in _AddressFieldEditor.cshtml_. The reason is that `Json.Serialize(object)` returns `IHtmlContent`, but `JConvert.SerializeObject` returns `string`. The latter is treated and user-facing text and gets HTML encoded. So you get invalid JS like this:
```javascript
commerceRegionsInitialize({&quot;US&quot;:{&quot;AL&quot;:&quot;Alabama&quot;,&quot;AK&quot;:&quot;Alaska&quot;, ... }});
```